### PR TITLE
Implements a getter for extracting rock_task_model from InputDataStream

### PR DESCRIPTION
**Pull request by hemker**
This feature is implemented on top of #23, so that one should be merged first.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,3 +5,5 @@ find_package(Rock)
 rock_activate_cxx11()
 rock_init(pocolog_cpp 0.1)
 rock_standard_layout()
+
+add_compile_options(-DBASE_LOG_NAMESPACE=${PROJECT_NAME})
```
Comment (2maz) [inline CMakeLists.txt:9]: Already part of [rock_init](https://github.com/rock-core/base-cmake/blob/7c60186292d3a18794ef6f7fb7d23e1f72ed5d56/modules/Rock.cmake#L130)

File: src/InputDataStream.cpp [hunk 2]
```diff
@@ -39,6 +40,23 @@ void InputDataStream::loadTypeLib()
...
+    std::map<std::string, std::string>::const_iterator it = desc.getMetadataMap().find(entry);
+    if(it == desc.getMetadataMap().end())
+    {
+        std::string errorMessage = "Error: Logfile does not contain metadata " + entry + ". Maybe old logfile?";
+        std::cout << errorMessage << std::endl << "MetaData : " << std::endl;
+        for(auto e : desc.getMetadataMap())
+        {
+            std::cout << e.first << " : " << e.second << std::endl;
+        }
+        throw std::runtime_error("Error: Logfile does not contain metadata CXXType. Maybe old logfile?");
```
Comment (2maz) [inline src/InputDataStream.cpp:54]: 1. why don't you reuse errorMessage here?
2. prefix Error does not hold much information if this is a runtime_error, better prefix with namespace, e.g., pocolog_cpp::InputDataStream::getMetadataEntry:

File: src/FileStream.cpp [hunk 3]
```diff
@@ -55,14 +56,14 @@ bool pocolog_cpp::FileStream::open(const char* fileName, std::ios_base::openmode
     writePos = 0;
     this->fileName = fileName;
     
-//     std::cout << "File opened " << (fd > 0) << " file size " << fileSize  << " blk size " << blockSize << std::endl;
+    LOG_DEBUG_S << "File opened " << (fd > 0) << " file size " << fileSize  << " blk size " << blockSize << std::endl;
```
Comment (2maz) [inline src/FileStream.cpp:59]: LOG_DEBUG_S adds a std::endl, so you could remove it here.

File: CMakeLists.txt (hunk 1 above)
Comment (hemker) [reply CMakeLists.txt]: Fixed

File: src/FileStream.cpp (hunk 3 above)
Comment (hemker) [reply src/FileStream.cpp:59]: All std::endls are removed, except from executables.

Comment (hemker) [reply src/FileStream.cpp]: All std::endls are removed now.

**Review by 2maz — APPROVED** [review]

**Pull request merged**